### PR TITLE
Added support for resolution configuration. Added option to start or not the conversion when opening ADC device.

### PR DESCRIPTION
**Pull request by daniel-p-carvalho**
Added option to start or not the conversion when opening ADC device.

### CODE REVIEW

File: arch/arm/src/stm32l4/stm32l4_adc.c [hunk 1]
```diff
@@ -1291,10 +1331,11 @@ static int adc_setup(FAR struct adc_dev_s *dev)
   adc_putreg(priv, STM32L4_ADC_SMPR1_OFFSET, ADC_SMPR1_DEFAULT);
   adc_putreg(priv, STM32L4_ADC_SMPR2_OFFSET, ADC_SMPR2_DEFAULT);
 
+  #ifdef HAVE_ADC_RESOLUTION
```
Comment (ghn-certi) [inline arch/arm/src/stm32l4/stm32l4_adc.c:1334]: Is this line formatted correctly?

Comment (daniel-p-carvalho) [reply arch/arm/src/stm32l4/stm32l4_adc.c:1334]: @ghn-certi Which line are you talking about? Are you referring to the coding standard?

Thanks

Comment (acassis) [reply arch/arm/src/stm32l4/stm32l4_adc.c:1371]: Hi @daniel-p-carvalho, I think @ghn-certi is talking about the "#ifdef" line, normally all #ifdef lines starts totally at left size even for indented code.

**Comment by xiaoxiang781216** [comment]
@daniel-p-carvalho could you squash PR into one or two patch?

**Comment by daniel-p-carvalho** [comment]
@xiaoxiang781216 how can I do this? I did a

$git rebase -i HEAD~5

but after this I have nothing to commit and nothing to push. I think that I am not doing it right. Can you help me?

Thanks

**Comment by davids5** [comment]
@daniel-p-carvalho in the editor you have to change the the `pick` to `s` to squash those commit into the one listed above.

**Comment by daniel-p-carvalho** [comment]
@davids5. I tried to squash the last 4 commits into one, keeping the first one. I hope that I was done this right.  Thanks for the help

**Comment by btashton** [comment]
Sill not quite right there are a bunch of merges. 

Try this:
```
git fetch origin
git rebase -i origin/master
```

This assumes that `origin` is the Apache remote and not your fork so adjust accordingly. Then select `s` instead of `pick` for the commits that you want to squash. You will then have to force push. 

It's a little funky at first but after you use interactive rebase a few time I suspect you will find it a very useful tool.

**Comment by daniel-p-carvalho** [comment]
Hi @btashton , thanks for the help.... I am moving this change to a new branch. Doing this I hope that the PR will be more clean and easy to review.

**Pull request closed without merging**